src/dymaptic.GeoBlazor.Core/Scripts/arcGisJsInterop.ts

```diff
@@ -2112,7 +2123,48 @@ export async function createLayer(layerObject: any, wrap?: boolean | null, viewI
             newLayer = kmlLayer;
             copyValuesIfExists(layerObject, kmlLayer, 'sublayers', 'blendMode', 'maxScale', 'minScale', 'title', 'visible');
             break;
-        default:
+        case 'wcs':
+            newLayer = new WCSLayer({
+                url: layerObject.url,
+                title: layerObject.title
+            });
+            let wcsLayer = newLayer as WCSLayer;
+            
+            if (hasValue(layerObject.renderer) && (layerObject.renderer.type == 'raster-stretch')) {
+                wcsLayer.renderer = buildJsRasterStretchRenderer(layerObject.renderer) as RasterStretchRenderer;
+
+                if (hasValue(layerObject.renderer.stretchType)) {
+                    wcsLayer.renderer.stretchType = layerObject.renderer.stretchType;
+                }
+                if (hasValue(layerObject.renderer.statistics)) {
+                    wcsLayer.renderer.statistics = layerObject.renderer.statistics;
+                }
+            }
+            if (hasValue(layerObject.multidimensionalDefinition) && layerObject.multidimensionalDefinition.length > 0) {
+                wcsLayer.multidimensionalDefinition = [];
+                for (let i = 0; i < layerObject.multidimensionalDefinition.length; i++) {
+
+                    let wcsMDD = new DimensionalDefinition;
+                    if (hasValue(layerObject.multidimensionalDefinition.VariableName)) {
+                        wcsMDD.variableName = layerObject.multidimensionalDefinition.VariableName;
+                    }
+                    if (hasValue(layerObject.multidimensionalDefinition.DimensionName)) {
+                        wcsMDD.dimensionName = layerObject.multidimensionalDefinition.DimensionName;
+                    }
+                    if (hasValue(layerObject.multidimensionalDefinition.Values)) {
+                        wcsMDD.values = layerObject.multidimensionalDefinition.Values;
+                    }
+                    if (hasValue(layerObject.multidimensionalDefinition.isSlice)) {
+                        wcsMDD.isSlice = layerObject.multidimensionalDefinition.isSlice;
+                    }
+                    wcsLayer.multidimensionalDefinition.push(wcsMDD);
+                }
+            }
+            copyValuesIfExists(layerObject, 'bandIds', 'copyright', 'coverageId', 'coverageInfo', 'customParameters', 'fields', 'interpolation', 'maxScale', 'minscale', 'rasterInfo');
```

Comment:
looking through these, I wonder if `fields` is the same type as used in other places. We do have a `buildJsField` method.

Reply:
it could be.  These were not essential to the building the sample page and following the pattern from other classes, these were included based on the properties in the rasterstretchrenderer